nomos-services/api/tests/todo.rs

```diff
@@ -165,13 +165,13 @@ mod todo {
     pub enum TodoError {
         /// Todo already exists conflict.
         #[schema(example = "Todo already exists")]
-        Conflict(String),
+        Conflict(Box<str>),
```

Comment:
Both String and Box<str> are heap allocated. What does it change and improve here?

Reply:
> A large enum used in several places can cause a negative runtime impact

`Box<str>` is made up of 2 words while `String` has 3 words. They have the same semantic and this conversion trick is commonly used in several projects.

Every time a function returns `Result<T, TodoError>`, the program allocates `max(size_of(T), size_of(TodoError)) + padding` bytes of stack memory, which can become a runtime problem if such a thing is a hot path called in several locations.

Of course, this particular scenario probably won't make such of a difference but it is about the message, the same message I tried to propagate trying to reduce the usage of third-party crates.

But again that is just me. If you want, I can revert this change.